### ISSUE TITLE
When some client has multiple instances, and we do remock, not all instances are updated

### ISSUE THREAD

**Issue by ellenaua** [dwyl/aws-sdk-mock#219]
**Issues**
1. When some client has multiple instances, and we do remock, not all instances are updated
2. When some client has multiple instances, and we do restore, not all instances are restored

**How to reproduce issue 1** 

```
// mock DynamoDB.getItem
awsMock.mock('DynamoDB', 'getItem', function(params, callback){
      callback(null, 'value1');
});

// you have multiple instances in your code
const ddb1 = new AWS.DynamoDB();
const ddb2 = new AWS.DynamoDB();
    
awsMock.remock('DynamoDB', 'getItem', function(params, callback){
   callback(null, 'value2');
});
```
Currently, only ONE of instances is updated in this case, and that is strange.
`ddb1.getItem()` returns value1, and `ddb2.getItem()` returns value2.

**How to reproduce issue 2** 

```
awsMock.mock('SNS', 'publish', function(params, callback){
      callback(null, 'message');
});

const sns1 = new AWS.SNS();
const sns2 = new AWS.SNS();

// publish function on both instances is mocked

st.equals(AWS.SNS.isSinonProxy, true);
st.equals(sns1.publish.isSinonProxy, true);
st.equals(sns2.publish.isSinonProxy, true);

 awsMock.restore('SNS', 'publish');

// only publish function on second instance (sns2) is restored
```

**Pull request with fix** 
I opened a pull request with fix https://github.com/dwyl/aws-sdk-mock/pull/218

**Comment by ellenaua** [comment]
@nelsonic could you please check my PR

**Comment by nelsonic** [comment]
@ellenaua apologies for the delay getting to this. ⌛ 
This will be a legit use case for some people. 👍 
Ideally we would also add to the Docs so people know it's possible to mock multiple instances. 📝 
Your PR has been merged. ✅ 

New version published to NPM: `aws-sdk-mock@5.4.0` 📦 
Invitation to be a maintainer of the repo sent. ✉️
